test/end-to-end/core.sh

```diff
@@ -426,6 +426,24 @@ os::cmd::expect_success "oc patch dc/failing-dc-mid -p '{\"status\":{\"latestVer
 os::cmd::expect_success_and_text 'oc logs --version=1 dc/failing-dc-mid' 'test mid hook executed'
 os::cmd::expect_success_and_text 'oc logs --previous dc/failing-dc-mid'  'test mid hook executed'
 
+os::log::info "Rollout latest"
+os::cmd::expect_success_and_text "oc tag --source docker centos:6 centos:latest" "centos:latest set to centos:6"
+os::cmd::try_until_text "oc get istag/centos:latest -o jsonpath --template={.image.dockerImageReference}" "centos@sha256:"
+INITIAL_CENTOS_SHA=$(oc get istag/centos:latest -o jsonpath --template={.image.dockerImageReference})
+os::cmd::expect_success_and_text "oc set image-lookup centos" 'imagestream "centos" updated'
+os::cmd::expect_success "oc run test --image=centos:latest --generator=deployment/apps.v1beta1 --command -- /bin/sleep infinity"
+os::cmd::expect_success_and_text "oc set triggers deploy/test --from-image=centos:latest -c test --manual" 'deployment "test" updated'
+oc::cmd::expect_failure_and_text "oc rollout latest deployment/test" "already runs the latest images"
```

Comment:
Nice.

Reply:
by accident this also exercise all the annotation trigger logic and lookup